lib/checkother.cpp

```diff
@@ -510,6 +511,8 @@ void CheckOther::checkRedundantAssignment()
                 // warn
                 if (hasCase)
                     redundantAssignmentInSwitchError(tok, nextAssign, tok->astOperand1()->expressionString());
+                if (isInitialization)
```

Comment:
I would like to have a `else if` here so we only warn about 1 problem at a time.